tastypie/resources.py

```diff
@@ -2054,6 +2065,35 @@ def filter_value_to_python(self, value, field_name, filters, filter_expr,
 
         return value
 
+    def get_query_terms(self, field_name):
+        """ Helper to determine supported filter operations for a field """
+
+        if field_name not in self.fields:
+            raise InvalidFilterError("The '%s' field is not a valid field" % field_name)
+
+        try:
+            django_field_name = self.fields[field_name].attribute
+            django_field = self._meta.object_class._meta.get_field(django_field_name)
+            if hasattr(django_field, 'field'):
+                django_field = django_field.field  # related field
+        except FieldDoesNotExist:
+            raise InvalidFilterError("The '%s' field is not a valid field name" % field_name)
```

Comment:
I'd like to see a test for this case (and the other negative case on line 2032, unrelated to this patch).  I can still merge this and take a stab at those cases (and some others) before cutting the next release, when I next get some time.  The related fields code has had lots of regressions in the past from insufficient tests.